main.cpp

```diff
@@ -0,0 +1,121 @@
+#include <iostream>
+#include <stdio.h>
+#include <stdlib.h>
+extern "C" {
+#include "mmio.h"
+}
+
+void matrix_vector(int rows, int cols, double* A, double* v, double* out) {
```

Comment:
You should call this `matrix_vector_product` as without the context of the assignment it is not clear what this function would do. This function also needs documentation that describes the inputs, size/shape of the pointers, etc.